contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -196,6 +196,28 @@ varbinaryin(PG_FUNCTION_ARGS)
 		 */
 		int			bc = (len - 1) / 2 + VARHDRSZ;	/* maximum possible length */
 
+		if (typmod >= (int32) VARHDRSZ && bc > typmod)
+		{
+			/* Verify that extra bytes are zeros, and clip them off */
+			char	*temp_result;
+			size_t	i;
+
+			temp_result = palloc0(bc);
+			bc = babelfish_hex_decode_allow_odd_digits(inputText + 2, len - 2, temp_result);
```

Comment:
If input length is less than typmod then we are doing the computation again on line 221.

Reply:
Actually this is intentional as the first call is just for the verification purpose here in which we have to transform the whole input string to bytea to determine if we can truncate it. Then the second call will do the actual transformation with the correct length. This is what a typmod cast function will also do, which is that first the input function will convert the whole string to bytea and then typmod cast function will re-do it with a different length.